app.py

```diff
@@ -178,8 +178,27 @@ def _annotate_timeframes(self, mmif: Mmif, **parameters) -> Mmif:
         # next, validate labels in the input annotations
         src_labels = sqh.validate_labelset(tps)
 
+        # validate user-provided tfLabelMap keys against actual TimePoint labels
+        if parameters['tfLabelMap']:
+            user_map_keys = set(parameters['tfLabelMap'].keys())
+            src_label_set = set(src_labels)
+            invalid_keys = user_map_keys - src_label_set
+            unmapped_keys = src_label_set - user_map_keys
+            if invalid_keys or unmapped_keys:
+                warning_msg = "tfLabelMap validation issues detected:"
+                if invalid_keys:
+                    warning_msg += f" Invalid label(s) in tfLabelMap (not in TimePoint labels): {sorted(invalid_keys)}."
```

Comment:
Use `TimePoint labelSet` precisely, then also mention that these will be removed when generating  `TimeFrame` annotations.